dsalgo/sort.py

```diff
@@ -18,169 +18,205 @@ def __new__(self, array, algo, reverse=False):
+        for j in range(0, n - i - 1):
+            if array[j] > array[j + 1]:
+                array[j], array[j + 1] = array[j + 1], array[j]
+                swap = 1
+        if swap == 0:
             break
-    if reverse==True:
+    if reverse:
         return array[::-1]
     return array
 
-def merge(array,reverse=False):
-
 
+def merge(array, reverse=False):
     """
     1.Divide:
-    If q is the half-way point between p and r, then we can split the subarray A[p..r] 
+    If q is the half-way point between p and r,
+    then we can split the subarray A[p..r]
     into two arrays A[p..q] and A[q+1, r].
     2.Conquer:
-    In the conquer step, we try to sort both the subarrays A[p..q] and A[q+1, r]. 
-     If we haven't yet reached the base case,
-     we again divide both these subarrays and try to sort them.
+    In the conquer step, we try to sort both
+    the subarrays A[p..q] and A[q+1, r].
+    If we haven't yet reached the base case,
+    we again divide both these subarrays
+    and try to sort them.
     3.Combine:
-    When the conquer step reaches the base step and we get two sorted subarrays A[p..q] and A[q+1, r] for array A[p..r],
-    we combine the results by creating a sorted array A[p..r] from two sorted subarrays A[p..q] and A[q+1, r].
+    When the conquer step reaches the base step and
+    we get two sorted subarrays A[p..q] and A[q+1, r] for array A[p..r],
+    we combine the results by creating a sorted array A[p..r]
+    from two sorted subarrays A[p..q] and A[q+1, r].
+     args:
+      :array:(list) -> list to be sorted
+      :reverse:(boolean) -> default = False,
+                            can be True for sort
+                            in reverse order
     """
-    
-    if len(array) >1: 
-        mid = len(array)//2  # mid
-        left = array[:mid]   # Dividing the array elements  
-        right = array[mid:]  # into 2 halves 
-  
-        merge(left)        # Sorting the first half 
-        merge(right)       # Sorting the second half 
+
+    if len(array) > 1:
+        mid = len(array) // 2  # mid
+        left = array[:mid]  # Dividing the array elements
+        right = array[mid:]  # into 2 halves
+
+        merge(left)  # Sorting the first half
+        merge(right)  # Sorting the second half
         i = j = k = 0
-          
-        # Copy data to left[] and right[] 
-        while i < len(left) and j < len(right): 
-            if left[i] < right[j]: 
-                array[k] = left[i] 
-                i+= 1
-            else: 
-                array[k] = right[j] 
-                j+= 1
-            k+= 1
-          
-        # Checking if any element was left 
-        while i < len(left): 
-            array[k] = left[i] 
-            i+= 1
-            k+= 1
-          
-        while j < len(right): 
-            array[k] = right[j] 
-            j+= 1
-            k+= 1
-    if reverse==True :
-     return array[::-1]
+
+        # Copy data to left[] and right[]
```

Comment:
what is left and right? Use self-explanatory names as much as possible

Reply:
> what is left and right? Use self-explanatory names as much as possible

I thought this is relevant because we splitting the array into halves so operations can be understood easily also there are single-line comments for further explanation.

Reply:
Then use names like left_half and write_half